plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/AppEngineLibraries.java

```diff
@@ -16,29 +16,44 @@
 
 package com.google.cloud.tools.eclipse.appengine.libraries;
 
-import java.util.Arrays;
-import java.util.Collections;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.RegistryFactory;
 
 import com.google.cloud.tools.eclipse.appengine.libraries.model.Library;
+import com.google.cloud.tools.eclipse.appengine.libraries.model.LibraryFactory;
+import com.google.cloud.tools.eclipse.appengine.libraries.model.LibraryFactoryException;
 
 public class AppEngineLibraries {
 
-  // TODO obtain libraries from extension registry
-  // https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/819
+  private static final Logger logger = Logger.getLogger(AppEngineLibraries.class.getName());
+  
+  // TODO obtain libraries by group name
+  // TODO cache library list
+  /**
+   * @return the three core appengine libraries
+   */
   public static List<Library> getAvailableLibraries() {
```

Comment:
They're both pulling from the plugin.xml/configuration elements for com.google.cloud.tools.eclipse.appengine.libraries
I.e. the definitions are centralized already. Do we need an additional service for this?